### PR TITLE
Fix PHP notice when outside of orders screens

### PR DESCRIPTION
**Pull request by jorgeatorres**
### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

After merging #38617 I _noticed_ a PHP notice started appearing on all screens:

Screenshot 2023-06-07 at 16 15 49

While #38617 made this evident, this is IMHO a failure in our orders page controller, since its logic is executed on every page even when not viewing an orders-related screen.

This PR addresses this by returning early if we're not on an admin page with `wc-orders` in its slug.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Check out `trunk`.
1. Visit any page outside of the Orders screen.
1. Confirm that you see the PHP notice mentioned above:
   ```Function Automattic\WooCommerce\Internal\Admin\Orders\PageController::is_order_screen was called incorrectly. is not a valid order type
   ```
1. Visit WC > Orders and confirm that the notice no longer appears.
1. Repeat the above steps on this branch, and confirm that the notice does not appear at any point.

### CODE REVIEW

File: plugins/woocommerce/phpstan.neon.dist
Comment (coreymckrill) [inline plugins/woocommerce/phpstan.neon.dist:1]: This seems like an unintentional inclusion.

File: plugins/woocommerce/phpstan.txt
Comment (coreymckrill) [inline plugins/woocommerce/phpstan.txt:1]: Same with this one.

**Review by coreymckrill — DISMISSED** [review]
👍 Approving, but the extraneous files should be removed before merging. Thanks again for catching this notice @jorgeatorres 

I also noticed from the PR screenshot that I forgot to update the version placeholder in the two `wc_doing_it_wrong` calls (currently `x.x.x`). Would it make sense to update those here? If not, I'll just open a separate PR.

**Comment by jorgeatorres** [comment]
Hey @coreymckrill! Thanks for the review. I did push some files by mistake, but those have been removed now. I've also updated the 'x.x.x' version numbers per your request.

You'll have to review again, though, as the change dismissed the previous review. Thanks again!

**Review by coreymckrill — APPROVED** [review]

**Pull request merged**
